tests/mockServer/mock.js

```diff
@@ -1320,6 +1320,20 @@ function getArtifacts(req, res) {
     })
   }
 
+  if (req.query['partition-by']) {
+    const uniqueObjects = {}
+
+    collectedArtifacts.forEach(artifact => {
+      const name = artifact.spec?.db_key ?? artifact.db_key
+      const lastUpdate = new Date(artifact.metadata.updated)
+
+      if (!uniqueObjects[name] || new Date(uniqueObjects[name].metadata.updated) < lastUpdate) {
+        uniqueObjects[name] = artifact
+      }
+    })
+    collectedArtifacts = Object.values(uniqueObjects)
+  }
```

Comment:
convert it to the common function that will be used to get partitioned result. Utilize lodash here